src/output.cpp

```diff
@@ -176,8 +182,25 @@ void print_colored_text( const catacurses::window &w, int y, int x, nc_color &co
     }
 }
 
-void trim_and_print( const catacurses::window &w, int begin_y, int begin_x, int width,
-                     nc_color base_color, const std::string &text )
+int print_colored_text( const catacurses::window &w, int y, int x, nc_color &cur_color,
+                        nc_color base_color, const std::vector<std::string> &text, size_t last_line /* = 0*/ )
+{
+    int width = getmaxx( w );
+    int height = getmaxy( w ) ;
+    if( ( y < height && y >= 0 ) && ( x < width && x >= 0 ) ) {
+        last_line = last_line > height ? height : last_line;
+        for( auto &str : text ) {
+            if( y < last_line ) {
```

Comment:
Something has to be done here with types too - I won't do code suggestion as it is probably not a one line change.

```
src/output.cpp: In function 'int print_colored_text(const catacurses::window&, int, int, nc_color&, nc_color, const std::vector<std::__cxx11::basic_string<char> >&, size_t)':
src/output.cpp:191:31: error: comparison of integer expressions of different signedness: 'size_t' {aka 'long long unsigned int'} and 'int' [-Werror=sign-compare]
         last_line = last_line > height ? height : last_line;
                     ~~~~~~~~~~^~~~~~~~
src/output.cpp:193:19: error: comparison of integer expressions of different signedness: 'int' and 'size_t' {aka 'long long unsigned int'} [-Werror=sign-compare]
             if( y < last_line ) {
                 ~~^~~~~~~~~~~
cc1plus.exe: all warnings being treated as errors
make: *** [Makefile:813: objwin/tiles/output.o] Error 1
```